### DISCUSSION TITLE
working-on-two-git-branches-at-once-with-git-worktree/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#46] [Announcements]
# working-on-two-git-branches-at-once-with-git-worktree/

In this post I describe how you can use git worktree to check out multiple branches at once, and some scenarios in which this can be useful

https://andrewlock.net/working-on-two-git-branches-at-once-with-git-worktree/

**Comment by solo12zw74** [comment]
Thanks for so detailed description. I am actively using this git feature for a long time and can approve, that it is much better than multiple cloning. And for me it is helpful to name worktree by template "RepositoryName-featureName", so repositories stored in the single "repos" folder, and Prefix define the repository and the suffix defines the feature.
Typically for me in person it is enouf to keep living 3-4 worktrees. One, the "main" worktree, folder with the repo name without any suffix , most actively used worktree (for example, new product feature "in work" are located here). #2 is a "review" worktree - here might be cheked out many branches for review or for helping colleagues, without need to switch in main worktree. It has suffix "-review" additionally to the repo name itself. And #3 usually is a "long-playing refactoring or research" worktree. It has the name of feature on research.

**Comment by MichalGrzegorzak** [comment]
Thank you, had no idea of such feature!

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at April 11 2022, 09:59, in reply to [Alexandre Konioukhov](https://disqus.com/by/alexandrekonioukhov/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/46#discussioncomment-6063049)</em>

---

<p>Yes and no. AFAIK, you can't create a worktree from VS, but you can open a sln in a linked worktree, and it will "see" the git repo correctly.</p>

**Comment by sarbash** [comment]
Thank you a lot. I never used this feature before but now I'm happy with it. My current project made me need it and gooogling lead me here. Btw VS Code + GitLens allow use the feature solely in GUI. Nevertheless I personally still think that cli is faster taking into count the embedded terminal of VS Code.
